lib/action_client/base.rb

```diff
@@ -20,11 +24,13 @@ def default(options)
         end
       end
 
+      def after_submit(with_status: nil, &block)
+        middleware.unshift ActionClient::Callbacks::AfterSubmit, with_status, &block
```

Comment:
What do you think about the `#prepend` alias for `#unshift`? I find it hard to remember which end `#unshift` operates on, whereas `#prepend` makes it explicit.

Reply:
Unfortunately, [`ActionDispatch::MiddlewareStack`](https://api.rubyonrails.org/classes/ActionDispatch/MiddlewareStack.html) does not declare that alias. I think we're stuck with `#unshift`.

Reply:
Oops, sorry, since it quacked like an `Array` I thought it _was_ an `Array`.